app.js

```diff
@@ -89,224 +59,40 @@ if (!process.send) {
 	winston.info('');
 }
 
-
 if (nconf.get('setup') || nconf.get('install')) {
-	setup();
+	require('./src/cli/setup').setup();
 } else if (!configExists) {
 	require('./install/web').install(nconf.get('port'));
 } else if (nconf.get('upgrade')) {
-	upgrade();
+	require('./src/cli/upgrade').upgrade(true);
 } else if (nconf.get('reset')) {
-	async.waterfall([
-		async.apply(require('./src/reset').reset),
-		async.apply(require('./src/meta/build').buildAll),
+	var options = {
+		theme: nconf.get('t'),
+		plugin: nconf.get('p'),
+		widgets: nconf.get('w'),
+		settings: nconf.get('s'),
+		all: nconf.get('a'),
+	};
+
+	async.series([
+		async.apply(require('./src/cli/reset').reset, options),
+		require('./src/meta/build').buildAll,
 	], function (err) {
-		process.exit(err ? 1 : 0);
+		if (err) {
+			throw err;
```

Comment:
Is outputting a stack trace the best course of action here? Wouldn't a graceful error handle work better?

I say this knowing full well that the old code didn't handle it any better :smile:

Reply:
I think outputting a stack trace is always more useful than _just_ outputting an error message. If something is wrong, we should detect it before an error is thrown in the first place.